order/src/main/java/com/caldevsupplychain/order/service/OrderServiceImpl.java

```diff
@@ -1,79 +1,131 @@
-import com.caldevsupplychain.order.util.ItemMapper;
 import com.caldevsupplychain.order.util.OrderMapper;
 import com.caldevsupplychain.order.vo.OrderBean;
 import com.caldevsupplychain.order.vo.OrderStatus;
 import com.google.common.base.Preconditions;
-import lombok.AllArgsConstructor;
-import org.springframework.data.domain.Page;
-import org.springframework.data.domain.PageRequest;
-import org.springframework.stereotype.Service;
-
-import javax.transaction.Transactional;
-import java.util.List;
-import java.util.Optional;
 
 
+@Slf4j
 @Service
 @AllArgsConstructor
 public class OrderServiceImpl implements OrderService {
 
 	private OrderRepository orderRepository;
+	private ItemRepository itemRepository;
 	private OrderMapper orderMapper;
-	private ItemMapper itemMapper;
 
 	@Override
 	@Transactional
 	public OrderBean createOrder(OrderBean orderBean) {
 
 		Preconditions.checkState(orderBean != null, "Must have orderBean in creating order");
 
-		Order order = orderMapper.toOrder(orderBean, new CycleAvoidingMappingContext());
+		CycleAvoidingMappingContext context = new CycleAvoidingMappingContext();
+
+		Order order = orderMapper.toOrder(orderBean, context);
 
 		orderRepository.save(order);
 
-		return orderMapper.toBean(order, new CycleAvoidingMappingContext());
+		order.getItems().stream().forEach(item -> {
+			item.setOrder(order);
+		});
+
+		itemRepository.save(order.getItems());
+
+		return orderMapper.toBean(order, context);
 	}
 
 	@Override
 	@Transactional
 	public OrderBean updateOrder(OrderBean orderBean) {
 
+		System.out.println(org.hibernate.Version.getVersionString());
+
 		Order order = orderRepository.findOne(orderBean.getId());
 
-		Preconditions.checkState(orderBean != null, ErrorCode.ORDER_ID_NOT_FOUND);
+		Preconditions.checkState(orderBean != null, ErrorCode.ORDER_UUID_NOT_FOUND);
 
-		Optional.ofNullable(orderBean.getTotalPrice()).ifPresent(totalPrice -> order.setTotalPrice(totalPrice));
+		// update order part
+		Optional.ofNullable(orderBean.getSku()).ifPresent(sku -> order.setSku(sku));
 
 		Optional.ofNullable(orderBean.getOrderType()).ifPresent(orderType -> order.setOrderType(orderType));
 
 		Optional.ofNullable(orderBean.getOrderStatus()).ifPresent(orderStatus -> order.setOrderStatus(orderStatus));
 
-		Optional.ofNullable(orderBean.getItems()).ifPresent(itemBeans -> order.setItems(itemMapper.toItemList(itemBeans)));
+		Optional.ofNullable(orderBean.getCurrency()).ifPresent(currency -> order.setCurrency(currency));
+
+		Optional.ofNullable(orderBean.getTotalPrice()).ifPresent(totalPrice -> order.setTotalPrice(totalPrice));
+
+		Optional.ofNullable(orderBean.getShippingInstruction()).ifPresent(shippingInstruction -> order.setShippingInstruction(shippingInstruction));
+
+		Optional.ofNullable(orderBean.getOrderNote()).ifPresent(orderNote -> order.setOrderNote(orderNote));
+
+		// update item list part
+		Optional.ofNullable(orderBean.getItems()).ifPresent(itemBeanList -> {
```

Comment:
Don't use `ofNullable` on collections. `orderBean.getItems().stream()` is the correct way to operate on collections.

Reply:
Fixed